### PR TITLE
Extract out common QI code for Xaml to use.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
@mgoertz-msft You would now call QuickInfoUtilities.CreateQuickInfoItemAsync which shares the real code with our QI impl.  You can then take the QuickInfoItem returned and effectively do:

`item.Sections.SelectMany(s => s.TaggedParts).ToImmutableArray()` to just flatten it town to teh TaggedText array that you seem to use.

**Comment by mgoertz-msft** [comment]
Thanks, I was wondering how I would get the TaggedText out of the QuickInfoItem now. :)

File: src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs [hunk 1]
```diff
@@ -155,11 +156,8 @@ private static bool HasNoErrors(ImmutableArray<ISymbol> symbols)
             CancellationToken cancellationToken)
         {
             var descriptionService = workspace.Services.GetLanguageServices(token.Language).GetRequiredService<ISymbolDisplayService>();
-            var formatter = workspace.Services.GetLanguageServices(semanticModel.Language).GetRequiredService<IDocumentationCommentFormattingService>();
             var syntaxFactsService = workspace.Services.GetLanguageServices(semanticModel.Language).GetRequiredService<ISyntaxFactsService>();
 
-            var showWarningGlyph = supportedPlatforms != null && supportedPlatforms.HasValidAndInvalidProjects();
-
             var symbols = tokenInformation.Symbols;
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs:161]: logic that cares abotu 'token' or 'tokenInformation' stayed here.  The values we need from those are then just passed into the common helper which was moved to a Utilities class.

File: src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs [hunk 2]
```diff
@@ -171,159 +169,9 @@ private static bool HasNoErrors(ImmutableArray<ISymbol> symbols)
                     s2.Kind == SymbolKind.NamedType ? 1 : 0).ToImmutableArray();
             }
 
-            var groups = await descriptionService.ToDescriptionGroupsAsync(workspace, semanticModel, token.SpanStart, symbols, cancellationToken).ConfigureAwait(false);
-
-            var sections = ImmutableArray.CreateBuilder<QuickInfoSection>(initialCapacity: groups.Count);
-
-            var symbol = tokenInformation.Symbols.First();
-            if (tokenInformation.ShowAwaitReturn)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/QuickInfo/CommonSemanticQuickInfoProvider.cs:179]: this logic was moved, effectively unchanged.  we just now don't access token/tokenInformation/etc. anymore, but we use the values passed in.

File: src/Features/Core/Portable/QuickInfo/QuickInfoUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        public static async Task<QuickInfoItem> CreateQuickInfoItemAsync(
+            Workspace workspace,
+            ISymbolDisplayService descriptionService,
+            SemanticModel semanticModel,
+            TextSpan span,
+            ImmutableArray<ISymbol> symbols,
+            SupportedPlatformData? supportedPlatforms = null,
+            bool showAwaitReturn = false,
+            NullableFlowState flowState = NullableFlowState.None,
+            CancellationToken cancellationToken = default)
```
Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/QuickInfo/QuickInfoUtilities.cs:32]: @mgoertz-msft you can ignore the supportedPlatform/showAwaitReturn/flowState as i imagine you don't care about them.

File: src/Features/Core/Portable/QuickInfo/QuickInfoUtilities.cs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    {
+        public static async Task<QuickInfoItem> CreateQuickInfoItemAsync(
+            Workspace workspace,
+            SemanticModel semanticModel,
+            TextSpan span,
+            ImmutableArray<ISymbol> symbols,
+            SupportedPlatformData? supportedPlatforms = null,
+            bool showAwaitReturn = false,
+            NullableFlowState flowState = NullableFlowState.None,
+            CancellationToken cancellationToken = default)
```
Comment (dibarbet) [inline src/Features/Core/Portable/QuickInfo/QuickInfoUtilities.cs:31]: should cancellation token just be required?

**Review by dibarbet — APPROVED** [review]

File: src/Features/Core/Portable/QuickInfo/QuickInfoUtilities.cs (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/QuickInfo/QuickInfoUtilities.cs:31]: will do.

**Pull request merged**